cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/template/etl/transform/LogParserTransformTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.template.etl.transform;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.template.etl.api.Transform;
+import co.cask.cdap.template.etl.common.MockEmitter;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+
+public class LogParserTransformTest {
+  private static final Schema SCHEMA = Schema.recordOf(
+    "event",
+    Schema.Field.of("body", Schema.of(Schema.Type.STRING))
+  );
+
+  @Test
+  public void transformS3Log() throws Exception {
+    StructuredRecord record = StructuredRecord.builder(SCHEMA)
+      .set("body", "13a9f69e4a00effd6b4b891dcbcabef632ef9afe38cc8b0a9da7caf6fec8c152 repository.cask.co " +
+        "[22/Jan/2015:11:03:21 +0000] 70.42.248.20 - 60068D4CA00DCAE4 REST.GET.OBJECT " +
+        "ubuntu/precise/amd64/releases/dists/precise/releases/i18n/Translation-en " +
+        "\"GET /ubuntu/precise/amd64/      releases/dists/precise/releases/i18n/Translation-en HTTP/1.1\" " +
+        "403 AccessDenied 231 - 10 - \"-\" \"Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.  17)\" -")
+      .build();
+
+    StructuredRecord record1 = StructuredRecord.builder(SCHEMA)
+      .set("body", "13a9f69e4a00effd6b4b891dcbcabef632ef9afe38cc8b0a9da7caf6fec8c152 repository.cask.co " +
```

Comment:
in fact, it may be better to find a 'test' CLF record from somewhere, as opposed to using an actual one.